### DISCUSSION TITLE
Blinds motor

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#18921] [General]
https://www.amazon.com/gp/product/B0BMLFZ4CP/

I wanted to buy some of these but it's not clear what these might be running. 
Does anyone know of a similar, solar based blinds motor like this that is tasmota compatible?

**Comment by barbudor** [comment]
IMHO Unlikely using an ESP so not Tasmotable
ESP (and wifi) is not very solar/battery friendly

If you can find similar stuff with Zigbee that could possibly easier to integrate with a Tasmota based zigbee bridge

**Reply by dac2020** [reply]
Thank you. Just thought I should ask and confirm. I'll do that then.

**Reply by barbudor** [reply]
Randomly found this on Ali
https://aliexpress.com/item/1005002378230551.html

**Reply by dac2020** [reply]
I should look on that site more often. Thanks for that lead.

**Comment by dac2020** [comment]
Seems I need to ask something.

Description says;

"Solar Powered Zigbee Control Smart Blinds Drive Motor Tuya Motorized Chain Roller Control Shade Shutter Drive Motor Replacement"

I thought zigbee was it's own hardware/firmware, unrelated to tuya but I see tuya mentioned here.

**Reply by barbudor** [reply]
Zigbee is a standard defining some radio protocols, not a hardware/firmware
There are different chips (MCU + radio module) that can be used to support Zigbee from both radio and protocol. They need a proper firmware on it to implement the protocol. There are some opensource firmware, some closed source but freely available or totally closed solutions.
For example the Sonoff ZbBridge Pro uses
- an ESP32 as master CPU with Wifi support
- an Texas Instruments CC2652  for the Zigbee radio
- Z-Stack Zigbee Network Processor (ZNP) firmware from texas Instruments

I have an Eachen Zigbee Bridge also using an ESP32 but a different Zigbee chipset and matching firmware.
Different implementations may provide slighly different features or variation from the standard but may things should be there.


Tuya is a provider of OEM design, software and ecosystem for people who are manufacturing the devices and stick their brand on it.
They can provide complete turnkey product or pieces (CPU module, software, ...)
By doing that they also insure a minimum of interop between products like using the same Tuya app to control Wifi/Bluetooth devices from different manufacturer (as far as the manufacturer use the reference design as-is and don't add inventions of their own)


So for Zigbee, they also provide complete sets for manufacturers to build products on.
Including software implementation

This is where slight issues may happen as Tuya may freely interpret some of the Zigbee standards, or adds things on top
This shouldn't be a problem as with the Tasmota Zigbee Plugin if a device seems to use some strange implementation, it should be possible to put it back in line. Or there is nothing that a little piece of Berry rule wouldn't fix ;)

**Reply by dac2020** [reply]
Thanks so much for the detailed explanation. I'll re-read this a few times to try and better understand then maybe order my first non tuya but zigbee device to get some hands on time with one.

**Reply by sfromis** [reply]
To be clear, with Zigbee you do not have trouble with firmware replacement on devices, as this is not needed. The Sonoff ZbBridge Pro mentioned by barbudor is a coordinator device for the Zigbee network (basically a hub, of which you only use one), where you can run Tasmota to facilitate smarthome integration various ways.

**Reply by barbudor** [reply]
Which actually means to start your zigbee journey with such blind controller, you will also need such coordinator (or gateway in muggle wording) and flash tasmota on it

I've started using zigbee for motion sensors and now adding wireless buttons.
Zigbee is great for totally wireless devices. For anything that needs mains power, I prefer to stick to wifi devices. My personal choice.

**Reply by dac2020** [reply]
You have my interest in zigbee. Any chance you know of a url/article that might show me the way to start. Like a tutorial where someone is using the parts you've mentioned.

I can order a couple of those blinds and use this as my first steps in using zigbee.

**Reply by dac2020** [reply]
I was about to order the blinds control but noticed it's for chain. What I need is to turn the stick that opens and closes the blinds so I'll look some more.

![image](https://github.com/arendst/Tasmota/assets/9727841/f172a137-46a7-49d1-8b66-5e0857cec62d)
